lib/context.js

```diff
@@ -769,13 +770,15 @@ function _expandIri(activeCtx, value, relativeTo, localCtx, defined, options) {
     }
 
     // use mapping if prefix is defined
-    if(activeCtx.mappings.has(prefix)) {
-      const mapping = activeCtx.mappings.get(prefix);
+    const mapping = activeCtx.mappings.get(prefix);
+    if(mapping && mapping._prefix) {
       return mapping['@id'] + suffix;
     }
 
     // already absolute IRI
-    return value;
+    if(_isAbsoluteIri(value)) {
```

Comment:
Having a colon isn’t sufficient to make it an absolute (or compact) IRI. This final test ensure that only an absolute IRI will be returned, otherwise, if falls through.

Reply:
Ok.